jina/executors/indexers/keyvalue.py

```diff
@@ -28,7 +28,7 @@ class ReadHandler:
         def __init__(self, path):
             with open(path + '.head', 'rb') as fp:
                 tmp = np.frombuffer(fp.read(), dtype=np.int64).reshape([-1, 4])
-                self.header = {r[0]: r[1:] for r in tmp}
+                self.header = {r[0]: None if np.array_equal(tmp[0][1:], (-1, -1, -1)) else r[1:] for r in tmp}
```

Comment:
@florian-hoenicke Not sure I understand what this is doing

Reply:
Checks if elements from `[1:]` in the first row are `(-1, ...)` ? What would that mean?

Reply:
Oh I see those are the markers of having been deleted. Can't you just delete the key from the header completely and then rewrite it? It's much smaller than the body so it shouldn't be a performance hindrance.

Or at least make that `(-1, ...)` a constant